mypy/test/data/check-modules.test

```diff
@@ -61,7 +61,7 @@ class Bad: pass
 
 [case testImportWithinBlock]
 import typing
-if None:
+if 0:
```

Comment:
It was previously needed, but is no longer.  Changed this back.

We needed this previously because mypy now knows the body of `if None` will never execute (in the same way that the body of an `if isinstance(0, str)` would never execute) and ignores it completely.  I disabled this behavior when not in strict Optional mode, so the change is no longer needed.